src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -1002,23 +1007,27 @@ TEST(SpatialPoolerTest, testUpdateBoostFactors) {
   sp.setActiveDutyCycles(initActiveDutyCycles3);
   sp.updateBoostFactors_();
   sp.getBoostFactors(resultBoostFactors3.data());
-
+  ASSERT_EQ(resultBoostFactors3.size(), trueBoostFactors3.size());
   ASSERT_TRUE(check_vector_eq(trueBoostFactors3, resultBoostFactors3));
+  }
 
+  { //test4: global inh + inh radius + boost str
   Real32 initActiveDutyCycles4[] = {0.1f, 0.3f, 0.02f, 0.04f, 0.7f, 0.12f};
   Real32 initBoostFactors4[] = {0, 0, 0, 0, 0, 0};
-  vector<Real32> trueBoostFactors4 = {1.94773f, 0.263597f,   4.33476f,
-                                      3.549f,   0.00482795f, 1.59467f};
+  vector<Real32> trueBoostFactors4 = {54.598148, 7.38906, 121.51038,
```

Comment:
the bug here (in a newly enabled test) was comparing in `float` which resulted in rounding errors.